website/assets/apps/megamap/components/InfrastructureResult.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react'
+import SearchActions from '../actions/SearchActions';
+
+export default class InfrastructureResult extends Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      infrastructure_type: []
+    }
+  }
+  
+  handleRoad(){
+    this.setState({
+      infrastructure_type: [1]
```

Comment:
This id should come from the backend, instead of being hard-coded here, so that if it ever changes, the map search still works. It looks like the `SearchView` already has these data in its state, so we should be able to pass them to the `InfrastructureResult` component